File: app/src/main/java/com/water/datasample/FileActivity.java
package com.water.datasample;

import android.content.Context;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.water.datasample.helper.FileHelper;

import java.io.IOException;

public class FileActivity extends AppCompatActivity
        implements View.OnClickListener{
    private EditText etFileName;
    private EditText etContent;
    private Button btnSave;
    private Button btnClean;
    private Button btnRead;
    private Button btnDelete;
    private Button btnFileList;
    private Context _context;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_file);

        bindViews();
    }


    private void bindViews() {
        _context = getApplicationContext();

        etContent = (EditText) findViewById(R.id.etContent);
        etFileName = (EditText) findViewById(R.id.etFileName);
        btnClean = (Button) findViewById(R.id.btnClean);
        btnSave = (Button) findViewById(R.id.btnSave);
        btnRead = (Button) findViewById(R.id.btnRead);
        btnDelete = (Button) findViewById(R.id.btnDelete);
        btnFileList = (Button) findViewById(R.id.btnFileList);

        btnClean.setOnClickListener(this);
        btnSave.setOnClickListener(this);
        btnRead.setOnClickListener(this);
        btnDelete.setOnClickListener(this);
        btnFileList.setOnClickListener(this);
    }


    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btnClean:
                etContent.setText("");
                etFileName.setText("");
                break;
            case R.id.btnSave:
                FileHelper fhsave = new FileHelper(_context);
                String savename = etFileName.getText().toString();
                String savecontent = etContent.getText().toString();

                try {
                    fhsave.save(savename, savecontent);
                    Toast.makeText(_context, "数据写入成功", Toast.LENGTH_SHORT).show();
                } catch (Exception e) {
                    e.printStackTrace();
                    Toast.makeText(_context, "数据写入失败", Toast.LENGTH_SHORT).show();
                }

                break;
            case R.id.btnRead:
                String readcontent = "";
                FileHelper fhread = new FileHelper(_context);

                try {
                    String readname = etFileName.getText().toString();
                    readcontent = fhread.read(readname);
                } catch (IOException e) {
                    e.printStackTrace();
                }

                Toast.makeText(_context, readcontent == ""?"文件读取失败！":
                        readcontent, Toast.LENGTH_SHORT).show();

                break;
            case R.id.btnDelete:
                FileHelper fhdelete = new FileHelper(_context);
                boolean deletecontent = false;
                try {
                    String deletename = etFileName.getText().toString();
                    deletecontent = fhdelete.delete(deletename);
                } catch (Exception e) {
                    e.printStackTrace();
                }

                Toast.makeText(_context, deletecontent?"删除文件成功！":
                        "删除文件失败！", Toast.LENGTH_SHORT).show();

                break;
            case  R.id.btnFileList:
                FileHelper fhlist = new FileHelper(_context);
                String filename = "";

                try {
                    String[] fs = fhlist.fileList();

                    for (String file:fs) {
                        filename += file + "\n";
                    }
                } catch (Exception e) {
                    e.printStackTrace();
                }

                Toast.makeText(_context, filename == ""?"没有找到文件！":
                        filename, Toast.LENGTH_SHORT).show();

                break;
        }
    }
}

File: app/src/main/java/com/github/androidhappyclub/datasample/FileActivity.kt
/*
 * MIT License
 *
 * Copyright (c) 2024 AndroidHappyClub
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.github.androidhappyclub.datasample

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.ave.vastgui.tools.view.toast.SimpleToast
import com.ave.vastgui.tools.viewbinding.viewBinding
import com.github.androidhappyclub.datasample.databinding.ActivityFileBinding
import com.github.androidhappyclub.datasample.helper.FileHelper
import java.io.IOException

class FileActivity : AppCompatActivity(R.layout.activity_file) {

    private val binding by viewBinding(ActivityFileBinding::bind)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding.btnClean.setOnClickListener {
            binding.etContent.setText("")
            binding.etFileName.setText("")
        }

        binding.btnSave.setOnClickListener {
            val helper = FileHelper(this)
            val saveName = binding.etFileName.text.toString()
            val saveContent = binding.etContent.text.toString()
            try {
                helper.save(saveName, saveContent)
                SimpleToast.showShortMsg("数据写入成功")
            } catch (e: Exception) {
                SimpleToast.showShortMsg("数据写入失败")
            }
        }

        binding.btnRead.setOnClickListener {
            val helper = FileHelper(this)
            var readContent = ""
            try {
                val readName = binding.etFileName.text.toString()
                readContent = helper.read(readName)
            } catch (e: IOException) {
                e.printStackTrace()
            }
            SimpleToast.showShortMsg(if (readContent === "") "文件读取失败！" else readContent)
        }

        binding.btnDelete.setOnClickListener {
            val helper = FileHelper(this)
            var deleteContent = false
            try {
                val deleteName = binding.etFileName.text.toString()
                deleteContent = helper.delete(deleteName)
            } catch (e: Exception) {
                e.printStackTrace()
            }
            SimpleToast.showShortMsg(if (deleteContent) "删除文件成功！" else "删除文件失败！")
        }

        binding.btnFileList.setOnClickListener {
            val helper = FileHelper(this)
            var fileName = ""
            try {
                val fs = helper.fileList()
                for (file in fs) {
                    fileName += file + "\n"
                }
            } catch (e: Exception) {
                e.printStackTrace()
            }
            SimpleToast.showShortMsg(if (fileName === "") "没有找到文件！" else fileName)
        }
    }

}